fonttools-rs/src/tables/C2PA.rs

```diff
@@ -0,0 +1,251 @@
+/**
+ * @copyright 2023 Monotype Imaging Inc.
+ *
+ * @file C2PA.rs
+ *
+ * @brief C2PA OpenType extension font table
+ *
+ */
+use otspec::types::*;
+use otspec::{
+    DeserializationError, Deserialize, Deserializer, ReaderContext, SerializationError, Serialize,
+};
+use otspec_macros::tables;
+use std::str;
+
+/// The 'C2PA' OpenType extension tag.
+pub const TAG: Tag = crate::tag!("C2PA");
+
+// Defines the internal C2PA header record
+tables!(
+    C2PARecordInternal {
+        uint16 majorVersion
+        uint16 minorVersion
+        u32 activeManifestUriOffset
+        uint16 activeManifestUriLength
+        u32 c2paManifestStoreOffset
+        u32 c2paManifestStoreLength
+    }
+);
+
+/// A C2PA record, containing information about a current active manifest and/or
+/// an embedded manifest store.
+#[derive(Clone, Debug, PartialEq, Eq)]
+#[allow(non_snake_case)]
+pub struct C2PA {
+    /// Major version of the C2PA table record
+    pub majorVersion: uint16,
+    /// Minor version of the C2PA table record
+    pub minorVersion: uint16,
+    /// Optional URI to an active manifest
+    pub activeManifestUri: Option<String>,
+    /// Optional embedded manifest store
+    pub c2paManifestStore: Option<String>,
```

Comment:
This should probably use a reference to binary data stored as [u8] ?

And are you going to drop the `c2pa` prefix on this?  Maybe call it `manifestStore` ?

Reply:
Changed to a `Option<Box<[u8]>>` and added an accessor `get_manifest_store(&self) -> Option<&[u8]>` to address.